From aeaae2dd533fe6475dd02d858d982b9b22e6996d Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Wed, 18 Aug 2021 10:00:26 -0500
Subject: [PATCH 19/N] pacman.c: handle cr on stdin as well

Improves compatibility with Windows targets, specifically when using
powershell for string transforming package names and piping into
pacman as it's virtually impossible to send newline only terminated
strings through pipes

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 src/pacman/pacman.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/pacman/pacman.c b/src/pacman/pacman.c
index e398855a..a495fd86 100644
--- a/src/pacman/pacman.c
+++ b/src/pacman/pacman.c
@@ -1152,6 +1152,10 @@ int main(int argc, char *argv[])
 				if(line[nread - 1] == '\n') {
 					/* remove trailing newline */
 					line[nread - 1] = '\0';
+					if (line[nread - 2] == '\r') {
+						/* remove trailing carriage returns */
+						line[nread - 2] = '\0';
+					}
 				}
 				if(line[0] == '\0') {
 					/* skip empty lines */
-- 
2.32.0

